enhancements/ingress/global-options-enable-hsts.md

```diff
@@ -0,0 +1,298 @@
+```
+#### As a cluster administrator, I want to enable HSTS on a node-type basis (infra-only, worker-only, or both)
+(TBD - waiting for clarification on this request.  Routes aren't associated with node types.)
+#### As a cluster administrator, I want to enable HSTS for specific subdomains
+(TBD - dependency on wildcard DNS record/certificate, or just the `includeSubDomains`)
+#### As a cluster administrator, I want to disable HSTS per-route when it is enabled globally
+Add a route annotation to the route, setting the directive `max-age=0`, e.g.:
+```yaml
+metadata:
+  annotations:
+    haproxy.router.openshift.io/hsts_header: max-age=0
+```
+
+### Implementation Details/Notes/Constraints
+Implementing this enhancement requires changes in the following repositories:
+
+- openshift/router
+- openshift/cluster-ingress-operator
+
+### Risks and Mitigations
+As previously mentioned, use of the `includeSubDomains` directive may cause problems unless the user
+is aware of its encompassing implications.  To mitigate the risk of unexpected side effects, an analysis
+of the behavior of `includeSubDomains` will be performed and documented.
+
+Additionally, the direction of overrides on global versus per-route HSTS annotations will be documented,
+because it may not be clear to the user that a global setting cannot override a per-route setting.  Otherwise
+it would not be possible to use a per-route setting.
+
+## Design Details
+
+### Open Questions
+There are numerous TBD sections in this proposal.  It is currently a work in progress until further
+clarification is received.
+
+### Test Plan
+HAProxy and Router have unit test coverage; for this enhancement, the unit tests are
+expanded to cover the additional four goals and the new `ROUTER_DEFAULT_HSTS` environment variable propagation
+from the `IngressController` to the router.
+
+The operator has end-to-end tests; for this enhancement, add the following tests:
+
+#### Enable global HSTS
+1. Create an IngressController that enables global HSTS
+2. Open a connection to this route and send a request
+3. Verify that a response is received and that the headers include HSTS as configured in step 1
+
+#### Override global HSTS on an individual route
+1. Create an IngressController that enables global HSTS
+2. Create a route that disables global HSTS with an annotation
+3. Open a connection to this route and send a request
+4. Verify that a response is received and that the headers do not include HSTS as configured in step 2
+
+### Graduation Criteria
+N/A
+
+### Upgrade / Downgrade Strategy
+On upgrade, any previous per-route HSTS configuration remains in effect.
+
+If a cluster administrator applied any of the new HSTS configuration options
+in 4.8, then downgraded to 4.7, the HSTS configuration settings would be ignored
+except for the per-route HSTS configurations.  The administrator would be responsible
+for removing any per-route HSTS configurations that were no longer applicable.
```

Comment:
Might be worth mentioning that an end-users browser, depending on the HSTS max-age setting, may still remember to access routes via HTTPS on downgrade (that is, until the HSTS record expires in the user's browser).